forth/block.fs

```diff
@@ -1,9 +1,13 @@
 require io
 
-( three block buffers at $c000-$cbff )
-
-create bbi 0 , 0 c, \ buf block id's
-create dirty 0 , 0 c,
+( 11 block buffers at $a000-$cbff.
+  this may be excessive, let's
+  shrink it once v is replaced with
+  a block-based editor. )
+
+\ buf block id's
+create bbi   0 , 0 , 0 , 0 , 0 , 0 c,
+create dirty 0 , 0 , 0 , 0 , 0 , 0 c,
```

Comment:
if it stays an odd number, say 5, then any block editor that does shadow blocks with still be able to use all 5 buffers while you browse +2 and -2

Reply:
whoops nvm I should have read more, I see you're doing `#11 mod` please ignore.

Reply:
It is really nice that you take a look at things. It is maybe getting closer to first release. Test and documentation are still missing.

Reply:
Also to be done: update `EVALUATE`, `\` and `REFILL` to support blocks.

Reply:
Probably need to support:

- disk change
- release/delete blocks

Reply:
A big decision: what line length should `\` assume?

C64 default screen width is 40 characters.
Forth traditional line length is 64 characters.

It has been requested in #245 , it would be good if the text editor allows lines longer than screen width. 64 character line length would help with that.

On the other hand, 40 character line length would make better use of screen ergonomy. Nearly a full block would fit on the screen at the same time.

Reply:
I've avoided the problem by only using `(` in my blocks. FWIW my vote is 40 to (1) make editors simpler to implement and because (2) horizontal scrolling is kinda gross UX. It's definitely unorthodox though. Perhaps a poll is in order?

Reply:
I don't think a poll is needed, just need to work out the pros and cons. I agree 40-width makes a lot of sense for the reason you mention.

The strongest argument for 64 is that long strings can be useful, for example, it allows defining a `S"` string that is 40 characters long or more. I am not sure if there is a clean way to do the same with 40-width.

Reply:
Actually: I realize now, long lines is not a problem for blocks. When `LOAD`ing a block, `S"` will parse until the next `"`, no matter if it is on another line or not. 

It is a problem if the code is later copied from blocks to files, but then it can be resolved by putting the string on a long line.

I think altogether, that makes a strong case for 40-width.

Reply:
Unlike files, block LOAD is specified to use the entire block source as the input buffer, so words like `s"` and `(` should be able to span across lines portably if I understand correctly (if I don't understand correctly please let me know).

Another strong argument for 64 is to make it more likely to be able to directly reuse any Forth source already in 16x64 screens out there that have `\` comments. I'm personally ambivalent to that though.

Reply:
For the most part, I don't think porting from 16x64 to 25x40+24 would be a big problem. At least not compared to all the other problems when porting code from a Forth to another.